arcade/examples/sprite_health.py

```diff
@@ -1,230 +1,297 @@
 
         # Set the background color
-        arcade.set_background_color(arcade.color.AMAZON)
+        self.background_color = arcade.color.AMAZON
 
-    def on_draw(self):
-        """
-        Render the screen.
-        """
-
-        # This command has to happen before we start drawing
+    def on_draw(self) -> None:
+        """Render the screen."""
+        # Clear the screen. This command has to happen before we start drawing
         self.clear()
 
-        # Draw all the sprites.
-        self.coin_list.draw()
+        # Draw all the sprites
+        self.player_sprite.draw()
+        self.enemy_sprite.draw()
         self.bullet_list.draw()
-        self.player_list.draw()
-
-        for coin in self.coin_list:
-            coin.draw_health_number()
-            coin.draw_health_bar()
-
-        # Render the text
-        arcade.draw_text(f"Score: {self.score}", 10, 20, arcade.color.WHITE, 14)
+        self.bar_list.draw()
 
-    def on_mouse_motion(self, x, y, dx, dy):
-        """
-        Called whenever the mouse moves.
-        """
-        self.player_sprite.center_x = x
+        # Draw the text objects
+        self.top_text.draw()
+        self.bottom_text.draw()
 
-    def on_mouse_press(self, x, y, button, modifiers):
-        """
-        Called whenever the mouse button is clicked.
-        """
-        # Gunshot sound
-        arcade.play_sound(self.gun_sound)
-        # Create a bullet
-        bullet = arcade.Sprite(":resources:images/space_shooter/laserBlue01.png", SPRITE_SCALING_LASER)
+    def on_mouse_motion(self, x: float, y: float, dx: float, dy: float) -> None:
+        """Called whenever the mouse moves."""
+        self.player_sprite.position = x, y
 
-        # The image points to the right, and we want it to point up. So
-        # rotate it.
-        bullet.angle = 90
+    def on_update(self, delta_time) -> None:
+        """Movement and game logic."""
+        # Check if the player is dead. If so, exit the game
+        if self.player_sprite.health <= 0:
+            arcade.exit()
 
-        # Give the bullet a speed
-        bullet.change_y = BULLET_SPEED
+        # Increase the enemy's timer
+        self.enemy_timer += delta_time
 
-        # Position the bullet
-        bullet.center_x = self.player_sprite.center_x
-        bullet.bottom = self.player_sprite.top
+        # Update the player's indicator bar position
+        self.player_sprite.indicator_bar.position = (
+            self.player_sprite.center_x,
+            self.player_sprite.center_y + INDICATOR_BAR_OFFSET,
+        )
 
-        # Add the bullet to the appropriate lists
-        self.bullet_list.append(bullet)
+        # Call updates on bullet sprites
+        self.bullet_list.on_update(delta_time)
 
-    def on_update(self, delta_time):
-        """ Movement and game logic """
+        # Check if the enemy can attack. If so, shoot a bullet from the
+        # enemy to the player
```

Comment:
This change would be more grammatically correct and clearer.
```suggestion
        # enemy toward the player
```
Although `to` is shorter, it could be read as there being some certainty the bullet will hit.